### PR TITLE
Avoid creating new node processes by leveraging processAsync

### PR DESCRIPTION
**Pull request by benmccann**
Closes https://github.com/mihar-22/svelte-jester/issues/25

### CODE REVIEW

**Comment by sebastianrothe** [comment]
Do you think we can get rid of CJS completely and move all files to ESM ?

**Comment by sebastianrothe** [comment]
I think we need to enforce a version of Jest >= 27, because it won't run with Jest 26 anymore.

**Comment by benmccann** [comment]
> Do you think we can get rid of CJS completely and move all files to ESM ?

I'd be in favor of that. It seems unrelated to this PR and best to do in a new one though

**Comment by sebastianrothe** [comment]
> > Do you think we can get rid of CJS completely and move all files to ESM ?
> 
> I'd be in favor of that. It seems unrelated to this PR and best to do in a new one though

I'm so sorry, but I disabled most of the tests with one of my older commits :(

https://github.com/mihar-22/svelte-jester/pull/53/files#diff-f638d82910738e07f34f9ca5c6cb2a01c223570f9c2425ddf60f61e2070c169dR64

**Comment by benmccann** [comment]
> I'm so sorry, but I disabled most of the tests with one of my older commits :(

at least you're not alone in making that exact mistake... https://github.com/sveltejs/kit/issues/2069#issuecomment-891271711

**Comment by mihar-22** [comment]
PR LGTM! Feel free to merge when you're ready. Excited about this and also interested in moving to ESM in another PR if anyone is down for it. Happy to cut a release once we get that done.

**Comment by benmccann** [comment]
Cool. I guess the main thing is getting the tests re-enabled first. Is that something you're looking at @sebastianrothe ?

**Comment by sebastianrothe** [comment]
Yes, I have them running, but I had to switch everything to ESM. Don't know if that's okay, or if we should keep CJS as well.

**Comment by benmccann** [comment]
I'd say go ahead and switch it to ESM

**Comment by sebastianrothe** [comment]
Please have a look.

**Comment by sebastianrothe** [comment]
> I'd say go ahead and switch it to ESM

I guess someone could run the code through Babel/Bundler and output a CJS version of it.

File: package.json [hunk 1]
```diff
@@ -2,7 +2,10 @@
   "name": "svelte-jester",
   "version": "1.7.0",
   "description": "A Jest transformer for Svelte - compile your components before importing them into tests",
-  "main": "src/transformer.cjs",
+  "main": "src/transformer.js",
```
Comment (benmccann) [inline package.json:5]:
```suggestion
  "main": "src/transformer.js",
  "type": "module",
```

File: src/__tests__/transformer.test.js [hunk 2]
```diff
@@ -1,27 +1,100 @@
...
+ * @jest-environment jsdom
+ */
+import { readFileSync } from 'fs'
+import { dirname } from 'path'
+import { fileURLToPath } from 'url'
+import { jest } from '@jest/globals'
+
+import createTransformer from '../transformer'
+
+// Node API __dirname is missing in Jest
```
Comment (benmccann) [inline src/__tests__/transformer.test.js:11]: it's not Jest, but rather ESM
```suggestion
// Node API __dirname is missing in ESM
```

File: README.md [hunk 3]
```diff
@@ -150,24 +150,21 @@ Add the following to your Jest config
 
 ```json
 "transform": {
-  "^.+\\.js$": "babel-jest",
```
[the remark this answers is not in the document]
Comment (sebastianrothe) [reply README.md:153]: This section was about adding the preprocessor and schould not be babel/ts/esbuild specific.

File: package.json (hunk 1 above)
Comment (sebastianrothe) [reply package.json:5]: `type: module` should be there already https://github.com/mihar-22/svelte-jester/blob/master/package.json#L6

**Comment by benmccann** [comment]
LGTM

**Comment by mihar-22** [comment]
Thanks for the works on this PR @benmccann and @sebastianrothe! Released in `1.8.0` 🥳

**Comment by mihar-22** [comment]
Oh crap. I was meant to major bump...

**Comment by mihar-22** [comment]
Made a mess but all resolved and released in `2.0.0`.

**Pull request merged**